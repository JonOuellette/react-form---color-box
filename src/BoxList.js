import React, {useState} from 'react';
import NewBoxForm from './NewBoxForm';
import Box from './Box';

const BoxList = () => {
    
    const [boxes, setBoxes ] = useState([]);
    
    const addBox = newBox => {
        setBoxes(boxes => [...boxes, newBox]);
      };

      const removeBox = id => {
        setBoxes(boxes => boxes.filter(box => box.id !== id));
      };

    return (
        <div>
            <NewBoxForm createBox={addBox} />
            {boxes.map(({id, height, width, backgroundColor}) => (
                <Box
                    key = {id}
                    id = {id}
                    height = {height}
                    width = {width}
                    backgroundColor = {backgroundColor}
                    removeBox={() => removeBox(id)}
                />
            )
            )}
        </div>
    )
}

export default BoxList