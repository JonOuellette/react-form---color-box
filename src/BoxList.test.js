import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

// SmokeTest
it("renders without crashing", function() {
    render(< BoxList />)
} )

//SnapTest
it("matches snapshot", function() {
    const {asFragment} = render (<BoxList/>);
    expect (asFragment()).toMatchSnapshot();
});

//front-end tests

it("can add a box", function () {
    const {getByLabelText, getByText} = render(< BoxList/>);

    //simulating filling out form
    const heightInput = getByLabelText(/box height/i);
    const widthInput = getByLabelText(/box width/i);
    const backgroundColorInput = getByLabelText(/box color/i);

    fireEvent.change(heightInput, { target: { value: '100' } });
    fireEvent.change(widthInput, { target: { value: '100' } });
    fireEvent.change(backgroundColorInput, { target: { value: 'red' } });

    //simulating form submission
    fireEvent.click(getByText('Add Box'));

     // Check if the box is added
    const box = getByText('X').previousSibling;
    expect(box).toHaveStyle('height: 100px');
    expect(box).toHaveStyle('width: 100px');
    expect(box).toHaveStyle('background-color: red');

})

it('can remove a box', function () {
    const { getByLabelText, getByText, queryByText } = render(<BoxList />);

})
  