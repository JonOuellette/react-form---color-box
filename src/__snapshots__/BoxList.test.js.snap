// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="height"
      >
         Box Height(in pixels) 
      </label>
      <input
        id="height"
        name="height"
        placeholder="Box Height"
        type="text"
        value=""
      />
      <label
        for="width"
      >
         Box Width(in pixels) 
      </label>
      <input
        id="width"
        name="width"
        placeholder="Box Width"
        type="text"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Box Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Box Color"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
